File: Codeforces/Div 2/Codeforces Round 990/C_Swap_Columns_and_Find_a_Path.cpp
#include <bits/stdc++.h>
using namespace std;
using LL = long long;
const LL INF = 0x3f3f3f3f3f3f3f3f;
#define endl '\n'

int main() {
    ios::sync_with_stdio(false); cin.tie(nullptr); cout.tie(nullptr);
    int t, n, a, b;
    cin >> t;
    while (t--) {
        cin >> n;
        vector<vector<int>> grid(2, vector<int>(n));
        for (int i = 0; i < 2; ++i) {
            for (int j = 0; j < n; ++j) {
                cin >> grid[i][j];
            }
        }
        vector<pair<LL, LL>> blocks(n);
        for (int i = 0; i < n; ++i) {
            blocks[i] = {grid[0][i], grid[1][i]};
        }
        sort(blocks.begin(), blocks.end(), greater<pair<int, int>>());
        vector<vector<vector<LL>>> dp(n + 1, vector<vector<LL>>(n + 1, vector<LL>(2, -INF)));
        dp[0][0][0] = 0;
        for (int i = 1; i <= n; ++i) {
            for (int j = 0; j <= n; ++j) {
                for (int flag = 0; flag < 2; ++flag) {
                    if (j > 0) {
                        dp[i][j][flag] = max(dp[i][j][flag], dp[i - 1][j - 1][flag] + blocks[i - 1].first);
                    }
                    dp[i][j][flag] = max(dp[i][j][flag], dp[i - 1][j][flag] + blocks[i - 1].second);
                    if (j > 0 && flag == 0) {
                        dp[i][j][1] = max(dp[i][j][1], dp[i - 1][j - 1][0] + blocks[i - 1].first + blocks[i - 1].second);
                    }
                }
            }
        }
        LL ans = -INF;
        for (int j = 0; j <= n; ++j) {
            ans = max(ans, dp[n][j][1]);
        }
        cout << ans << endl;
    }
    return 0;
}

File: Codeforces/Div 2/Codeforces Round 990/C_Swap_Columns_and_Find_a_Path.py
"""
顯然有一部份的 col 會選擇上面，有另一部份的 col 會選擇下面，且恰有 1 個 col 同時選擇上面和下面
基於貪心思路，每個 col 選擇都先選擇較大的部分，即 max(col[0][i], col[1][i])
而剩下的較小的部分 min(col[0][i], col[1][i])，則只能在 n 個 col 中選擇 1 個，因此維護其最大的較小值
"""
t = int(input())

for _ in range(t):
    n = int(input())
    grid = [list(map(int, input().split())) for _ in range(2)]

    s = 0
    k = -float('inf')
    for j in range(n):
        s += max(grid[0][j], grid[1][j]) # 先選擇較大的部分
        k = max(k, min(grid[0][j], grid[1][j])) # 維護最大的較小值
    print(s + k)